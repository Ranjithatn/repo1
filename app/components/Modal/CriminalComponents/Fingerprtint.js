import React, { Fragment } from "react";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import NumericInput from "react-numeric-input";
import Modal from "../Modal";
import Label from "../../Label/Label";
import Image from "../../Image/Image";
import { reg_space } from "../../../utils/regEx";
// import "./UpdateCriminalModal.scss";

import { sortFingers } from "../../Tenprint/TenprintVerify/Helper";

function Fingerprtint({ formatMessage, JobType, LeftFingers, RightFingers }) {
  // console.log("Fingerprtint :: LeftFingers, RightFingers",LeftFingers, RightFingers);

  let sortedLeftFingers = LeftFingers;
  let sortedRightFingers = RightFingers;

  if (
    sortedLeftFingers &&
    sortedLeftFingers.fingers &&
    sortedLeftFingers.fingers.length > 0
  ) {
    sortedLeftFingers = sortFingers(LeftFingers);
  }

  if (
    sortedRightFingers &&
    sortedRightFingers.fingers &&
    sortedRightFingers.fingers.length > 0
  ) {
    sortedRightFingers = sortFingers(RightFingers);
  }

  // if ( sortedLeftFingers.fingers.length === 0 && LeftFingers.fingers.length > 0 ) { sortedLeftFingers = LeftFingers; }
  // if ( sortedRightFingers.fingers.length === 0 && RightFingers.fingers.length > 0 ) { sortedRightFingers = RightFingers; }

  // console.log("sortedLeftFingers",sortedLeftFingers);
  // console.log("sortedRightFingers",sortedRightFingers);

  return (
    <Fragment>
      {JobType !== "Custom" ? (
        <div className="tenprint-fields">
          <Label text={formatMessage({ id: "fingerprints" })} />
          <div className="columns">
            {sortedLeftFingers && sortedLeftFingers.length > 0 ? (
              <div className="column">
                {sortedLeftFingers.map((name, i) => {
                  return (
                    <Fragment>
                      <div
                        key={name + i}
                        className="segmented-data-container"
                        title={
                          name.annotation
                            ? name.annotation + ":" + name.annotationNote
                            : ""
                        }
                      >
                        {name.thumbNail !== "" ? (
                          <Image
                            src={"data:image/png;base64," + name.thumbNail}
                            id={name.position}
                          />
                        ) : (
                          <span title={name.annotationNote}>
                            {name.annotation}
                          </span>
                        )}
                      </div>
                      <span className="impression-name">
                        {name.position === "LeftRing" &&
                        JobType === "Latent" ? (
                          <span>{formatMessage({ id: "Unknown" })}</span>
                        ) : (
                          formatMessage({ id: name.position })
                        )}
                      </span>
                      {/* <Switch
                        id={name.position + i}
                        value={name.position}
                        className="is-primary is-rounded"
                      /> */}
                    </Fragment>
                  );
                })}
              </div>
            ) : (
              ""
            )}
            {sortedRightFingers && sortedRightFingers.length > 0 ? (
              <div className="column ">
                {sortedRightFingers.map((name, i) => {
                  return (
                    <Fragment>
                      <div
                        className="segmented-data-container"
                        title={
                          name.annotation
                            ? name.annotation + ":" + name.annotationNote
                            : ""
                        }
                      >
                        {name.thumbNail !== "" ? (
                          <Image
                            src={"data:image/png;base64," + name.thumbNail}
                            id={name.position}
                          />
                        ) : (
                          <span title={name.annotationNote}>
                            {name.annotation}s
                          </span>
                        )}
                      </div>

                      <span className="impression-name">
                        {formatMessage({ id: name.position })}
                      </span>
                      {/* <Switch
                        id={name.position + i}
                        value={name.position}
                        className="is-primary is-rounded"
                      /> */}
                    </Fragment>
                  );
                })}
              </div>
            ) : (
              ""
            )}
          </div>
          {/* <span className="noteLabel">
            {" "}
            {formatMessage({ id: "registerNote" })}{" "}
          </span> */}
        </div>
      ) : (
        ""
      )}
    </Fragment>
  );
}
export default Fingerprtint;
