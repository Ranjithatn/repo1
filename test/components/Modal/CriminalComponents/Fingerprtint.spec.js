import React from "react";
import toJson from "enzyme-to-json";
import Fingerprtint from "../../../../app/components/Modal/CriminalComponents/Fingerprtint";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

const props = {
  formatMessage: data => data.id,
  LeftFingers: { fingers: [] },
  RightFingers: { fingers: [] }
}

describe('<Fingerprtint />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Fingerprtint {...props} />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
