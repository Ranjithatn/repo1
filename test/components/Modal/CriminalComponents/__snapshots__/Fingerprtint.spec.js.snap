// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fingerprtint /> should render without crashing 1`] = `
<Fingerprtint
  LeftFingers={
    Object {
      "fingers": Array [],
    }
  }
  RightFingers={
    Object {
      "fingers": Array [],
    }
  }
  formatMessage={[Function]}
>
  <div
    className="tenprint-fields"
  >
    <Label
      text="fingerprints"
    >
      <label
        className="label"
        htmlFor=""
        name=""
        style={Object {}}
      >
        fingerprints
      </label>
    </Label>
    <div
      className="columns"
    />
  </div>
</Fingerprtint>
`;
